### PR TITLE
Auto-retry on transient embedding API errors

### LINKED ISSUE #6: Auto-retry on transient embedding API errors
## Summary
임베딩 API 호출 시 rate limit, timeout 등 일시적 에러에 대한 자동 재시도가 없음. 대규모 sync에서 한 번의 429 에러로 전체 sync가 실패함.

## Desired Behavior
- Rate limit (429), timeout, 5xx 에러에 대해 exponential backoff 재시도
- 최대 재시도 횟수 설정 가능 (기본 3회)
- 재시도 시 progress에 표시 (e.g. "retrying batch 3/64...")
- 영구 에러(401 auth, 400 bad request)는 즉시 실패

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### PR DESCRIPTION
**Pull request by vkehfdl1**
## Summary

Add exponential backoff retry for transient embedding API errors (429 rate limit, 5xx, timeout, connection errors) during sync, query, and verify --fix operations. Permanent errors (401 auth, 400 bad request, invalid API key) fail immediately without retry.

- `_is_transient_error()` classifies errors via duck-typing + string matching (no HTTP library imports)
- `_embed_with_retry()` / `_async_embed_with_retry()` with exponential backoff (1s, 2s, 4s... cap 30s)
- `sync()` gains `max_retries` param (resolves from arg → config → default 3)
- `--max-retries` CLI flag on sync subcommand
- 28 new tests covering all retry paths

## Test plan

- [x] All 253 tests pass (`make test`)
- [x] `make check` passes (lint, type checking, deptry)
- [x] `pytest tests/acceptance/test_retry.py -v` — 28 new tests pass

close #6

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: minsync/core.py [hunk 1]
```diff
@@ -1726,11 +1742,117 @@ def _remove_path(self, path: Path) -> None:
...
 def _parallel_embed_async(
     embedder: Any,
     texts: list[str],
     sub_batch_size: int,
     max_concurrent: int,
+    *,
+    max_retries: int = 3,
+    quiet: bool = False,
 ) -> list[list[float]]:
     """Run sub-batches of ``embedder.async_embed()`` in parallel via asyncio.
```
Comment (vkehfdl1) [inline minsync/core.py:1855]: **Code Review Debate** (2/3 reviewers flagged this)

:red_circle: **Devil's Advocate**: `_is_transient_error()` defaults unknown errors to transient (`return True` at the end). This means any programming bug (e.g. `TypeError`, `KeyError`) in the embedding pipeline will be silently retried 3 times before surfacing, adding unnecessary latency and masking real issues.

:white_circle: **Neutral Judge**: The default-to-transient choice is defensible for an embedding retry wrapper — the cost of one unnecessary retry is low vs. the cost of failing a large sync for a transient blip. However, consider logging at DEBUG level when falling through to the default case.

:green_circle: **Approval Advocate**: No concerns raised for this code. The docstring clearly documents this as an intentional design choice.

**Suggested Fix**: Acceptable as-is. Consider adding a debug-level log when the unknown fallback is used, to aid troubleshooting.
**Risk Level**: MEDIUM (Probability: LOW, Impact: LOW)

File: minsync/core.py [hunk 2]
```diff
@@ -802,9 +817,9 @@ def query(
         effective_filter = f"{base_filter} AND {normalized_filter}" if normalized_filter else base_filter
 
         try:
-            vectors = embedder.embed([query])
+            vectors = _embed_with_retry(embedder.embed, [query], max_retries=3, quiet=True)
```
Comment (vkehfdl1) [inline minsync/core.py:821]: **Code Review Debate** (2/3 reviewers flagged this)

:red_circle: **Devil's Advocate**: `query()` and `_repair_path()` hardcode `max_retries=3` instead of reading from config. Users who set `max_retries: 0` in config to disable retry will still get retries on query and verify --fix, creating inconsistent behavior.

:white_circle: **Neutral Judge**: Minor configuration inconsistency. For query (single text, fast), hardcoded 3 is pragmatic. For _repair_path (verify --fix), it could process many files so configurability matters more. Low probability this causes real issues since the default config is also 3.

:green_circle: **Approval Advocate**: No concerns raised for this code.

**Suggested Fix**: Acceptable as-is for now. If needed, a follow-up PR can thread `max_retries` through `query()` and `verify()` method signatures.
**Risk Level**: LOW (Probability: LOW, Impact: LOW)

**Comment by vkehfdl1** [comment]
## /refactor Code Review Summary

### Results

| Classification | Count | Action |
|----------------|-------|--------|
| Unanimous (3/3) | 0 | Auto-fixed |
| Majority (2/3) | 2 | Inline comment |
| Single (>= MEDIUM) | 0 | Inline comment |
| Single (LOW) | 3 | Skipped |
| **Total findings** | **5** | |

### Auto-Fixed Issues
No issues were unanimously identified for auto-fix.

### Inline Comments Posted
- `minsync/core.py:1858`: Unknown errors default to transient in `_is_transient_error()` (majority)
- `minsync/core.py:820`: Hardcoded `max_retries=3` in `query()` and `_repair_path()` (majority)

### Top Strengths (from Approval Advocate)
1. **Duck-typing error classification**: `_is_transient_error()` avoids HTTP library imports while supporting httpx, openai SDK, and other clients through attribute-based detection -- zero coupling to external dependencies.
2. **Comprehensive test coverage**: 28 new tests cover all retry paths (sync, async, query, verify --fix), edge cases (max_retries=0, permanent errors), and output behavior (stderr progress, quiet mode).
3. **Clean separation of concerns**: Retry logic is extracted into standalone module-level functions (`_embed_with_retry`, `_async_embed_with_retry`) that compose cleanly with existing embedding paths without modifying the core sync/query flow.

### Reviewer Verdicts
- :red_circle: **Devil's Advocate**: Default-to-transient fallback could mask bugs; hardcoded retry counts in query/verify create inconsistency with configurable sync retries.
- :white_circle: **Neutral Judge**: Well-structured retry implementation with appropriate tradeoffs. Minor config inconsistency is acceptable for v1. Low overall risk.
- :green_circle: **Approval Advocate**: Strong PR that solves a real production pain point. Clean code, thorough tests, follows existing patterns. Recommend approval.

---
*Generated by `/refactor` -- 3-agent code review debate*

File: minsync/core.py (hunk 2 above)
Comment (vkehfdl1) [reply minsync/core.py:821]: ignore

File: minsync/core.py (hunk 1 above)
Comment (vkehfdl1) [reply minsync/core.py:1855]: ignore
